### PR TITLE
refactor: docs: Docs Overhaul

### LINKED ISSUE #710: update "Integration Tests" document
## Possible Solution

The integration tests on this page needs to explained in [this page](https://hackmd.io/-bEre9tGTp-vkVsK7FjDpg)

## Checklist

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #867: update “Configuration” document
## Possible Solution

- Please review the contents of [this page ](https://hackmd.io/lV7lEQ-GR6SZjdMQv3Jbgg) and give feedback on this ticket

## Checklist

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #868: update “Command-Line” document
## Possible Solution

- update “Command-Line” document to include latest changes
- separate and distinguish `local flags` from `global flags` in command output
- see existing page at https://chainsafe.github.io/gossamer/command-line/

## Checklist

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #918: update "Host Architecture" document
## Possible Solution

- Review and amend the reference doc page at this [link](https://hackmd.io/11CbS9ClT4mTZM0FK34ItQ)

## Checklist

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #1213: Documentation - Gossamer Lifecycles
Based on the Gossamer Onboarding presentations, create diagrams and associated descriptions of the various lifecycles in Gossamer.

### LINKED ISSUE #1214: Documentation - Gossamer Architecture
Based on the Gossamer Onboarding presentations, updated the existing architecture document & diagrams to accurately reflect Gossamer today.

### LINKED ISSUE #1357: Documentation
## Issue summary

- Improve and refactor public documentation for Gossamer
- Should include detailed instructions on getting started, architecture, running the node, cli, rpc, and different packages 

## Other information and links

-

### PR DESCRIPTION
**Pull request by RyRy79261**
## Review instructions
The contents of each page can be edited at the HackMD link attached to each, for where said files reside in this branch, the path is provided

## Getting Started 
  - [x] [Overview](https://hackmd.io/bMsZctpUThenOo7lLhBZ_g) - docs/docs/index.md
  - [x] [Installation](https://hackmd.io/3umymBN9TvumSLaXXNY9PQ) - docs/docs/getting-started/installation.md
  - [x] [Host Architecture](https://hackmd.io/11CbS9ClT4mTZM0FK34ItQ)  - docs/docs/getting-started/overview/host-architecture.md
  - [x] [Package lib](https://hackmd.io/LfzVBbRPSYW7tgju0ZicwA) - docs/docs/getting-started/overview/package-library.md
## Resources
  - [x] [General Resources](https://hackmd.io/-FikBPYKTYKMy0uf4iLZqA) - docs/docs/getting-started/resources/general-resources.md
  - [x] [Developer Resources](https://hackmd.io/UleqqjshQCe8gk-TiT8u_Q) - docs/docs/getting-started/resources/developer-resources.md
## Usage
  - [x] [Command line](https://hackmd.io/tBlAYz3dSw2pSlZS_6Lcuw) - docs/docs/usage/command-line.md
  - [x] [Configuration](https://hackmd.io/lV7lEQ-GR6SZjdMQv3Jbgg) - docs/docs/usage/configuration.md
  - [x] [Import Runtime](https://hackmd.io/U17mRva-QLmN0bOhi9vsHw) - docs/docs/usage/import-runtime.md
  - [x] [Import State](https://hackmd.io/gnG8DCHTR0-JjT1Vn6N8AA?both) - docs/docs/usage/import-state.md
## Integrate
  - [x] [Connect polkadot.js](https://hackmd.io/cQqRdaP-Ql2i-stRoVlhEA) - docs/docs/integrat/connect-to-polkadot-js
## Testing & Debugging @dutterbutter 
  - [x] [Running intergration tests](https://hackmd.io/-bEre9tGTp-vkVsK7FjDpg) - docs/docs/testing-and-debugging/intergration-tests.md
  - [x] [Running unit tests](https://hackmd.io/HKIJbdc0T0uaxbebFy5Dpw) - docs/docs/testing-and-debugging/unit-tests.md
  - [x] [Logger usage](https://hackmd.io/qxsxNhGQQGWYTLfMz5_Oyw) - docs/docs/testing-and-debugging/logger-usage.md
  - [x] [Debugging](https://hackmd.io/bfVYAP2gTe6bjv9TMJ8tqg) - docs/docs/testing-and-debugging/debugging.md
## Deployment
  - [x] [Docker usage](https://hackmd.io/mVQkUBovRFmHgMvbESfwsA) - docs/docs/deployment/docker-usage.md
## Advanced
  - [x] [SCALE](https://hackmd.io/RRIH_vwvQeG84yAGwQEIMw) - docs/docs/advanced/scale-examples.md
## Contributing
  - [x] [Overview](https://hackmd.io/neZBYwoQRlWd-r_KRew__w) - docs/docs/contibuting.md

## Tests

```

```

## Checklist

- [ ] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [ ] I have provided as much information as possible and necessary
- [ ] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1357
- closes #1214
- closes #1213
- closes #1357
- closes #918 
- closes #868
- closes #867
- closes #710

### CODE REVIEW

File: docs/docs/integrate/connect-to-polkadot-js.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+bin/gossamer --rpc --ws --wsport 8546 --rpcmods system,author,chain,state,account,rpc --key alice
+```
+
+### Connecting the app to your node
+
+You'll need to setup the polkadot.js/apps to use a custom endpoint to connect to your gossamer node.  Open [polkadot.js.org/apps](https://polkadot.js.org/apps).
+
+Once you've opened the app in your browser, you should see it connected to the Polkadot network: 
+
+<img src="/assets/tutorial/connect-1.png" />
```
Comment (dutterbutter) [inline docs/docs/integrate/connect-to-polkadot-js.md:24]: Should these images be included in the repo or any other suggestion?

**Comment by noot** [comment]
@RyRy79261 can you remove `Run official nodes`? I don't think it's needed

File: docs/docs/integrate/connect-to-polkadot-js.md (hunk 1 above)
Comment (RyRy79261) [reply docs/docs/integrate/connect-to-polkadot-js.md:24]: @dutterbutter Usually I would have them in the repo for build purposes but hosting them somewhere is also an option

**Comment by dutterbutter** [comment]
@RyRy79261 Can you update the Overview section, it shouldn't contain our readme badges, and there is no need to have the outdated table of contents there.

**Comment by noot** [comment]
@RyRy79261 could you add this to the docs as well? https://hackmd.io/@nZ-twauPRISEa6G9zg3XRw/SJ8BgXOXO

**Comment by dutterbutter** [comment]
@RyRy79261 List of items to update / change. Please also make sure each section is up to date with the linked above HackMD pages. 

### **Getting Started Section**
- [ ] Remove table of contents in overview section
- [ ] Update initial overview section, for example this section 

`For more information about Gossamer, the Polkadot ecosystem, and how to use Gossamer to build and run nodes for various blockchain protocols within the Polkadot ecosystem, check out the Gossamer Docs.
`

- [ ] General resources link is broken
- [ ] Get Started link is broken
- [ ] Configuration link is broken
- [ ] import runtime link is broken
- [ ] custom services link is broken

**Host architecture**
- [ ] Remove TODO: update "Host Architecture" document #918
- [ ] Custom services link is broken
- [ ] Add embedded link when referencing BABE and GRANDPA
- [ ] Add link to JSON-RPC PSP under RPC service
- [ ] Include section for block production and add diagram and dialogue 

**Package libs**
- [ ] Update all referenced links as they seem to be broken
- [ ] Fill out Test packages section

**Installation**
- [ ] Update to include docker installation in hackmd links

**General Resources**
- [ ] terms link is broken
- [ ] Please add latest Gossamer blog links

**Developer Resources**
- [ ] Fix parachain list format
- [ ] Fix block production & consensus list format 

### **Usage Section**

**Configuration**
- [ ] Remove or add TODO: How to target & use custom config

**Import Runtime**
- [ ] The word containing has a typo under 1. Create genesis spec file with custom runtime section
- [ ] Can you add a screenshot of a running node to illustrate at the end

### Integrate
- [ ] Add in the polkadot.js material 
- [ ] Once Telemetry is added please include here as well

### Testing 
- [ ] Please include the testing material created

### Deployment
- [ ] You may remove at this time

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -1,56 +1,23 @@
...
-          git add --all
-          git -c user.name="${GITHUB_ACTOR}" -c user.email="${GITHUB_ACTOR}@users.noreply.github.com" \
-            commit --quiet \
-            --message "Deploy docs from ${GITHUB_SHA}" \
-            --message "$SOURCE_COMMIT"
-          git push
-          popd &>/dev/null
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # CUSTOM_DOMAIN: optionaldomain.com
```
Comment (dutterbutter) [inline .github/workflows/docs.yml:20]: Add our custom domain here please  https://gossamer.chainsafe.io/

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -1,56 +1,23 @@
...
+          CONFIG_FILE: docs/mkdocs.yml
+          EXTRA_PACKAGES: build-base
+          # GITHUB_DOMAIN: github.myenterprise.com
```
Comment (dutterbutter) [inline .github/workflows/docs.yml:23]: Remove if not needed

File: docs/mkdocs.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+# Configuration
+markdown_extensions:
+  - meta
+  - admonition
+  - codehilite
+  - toc:
+      permalink: true
+
+theme:
```
Comment (dutterbutter) [inline docs/mkdocs.yml:18]: Could you add the ability to toggle between light and dark mode? Guide here: https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/

Code snipped (need to update colors accordingly):
```
theme:
  palette:

    # Light mode
    - media: "(prefers-color-scheme: light)"
      scheme: default
      primary: indigo
      accent: indigo
      toggle:
        icon: material/toggle-switch-off-outline
        name: Switch to dark mode

    # Dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      primary: blue
      accent: blue
      toggle:
        icon: material/toggle-switch
        name: Switch to light mode
```

File: docs/mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    - Configuration: ./usage/configuration.md
+    - Import Runtime: ./usage/import-runtime.md
+    - Import State: ./usage/import-state.md
+  - Integrate:
+    - Connect to Polkadot.js: ./integrate/connect-to-polkadot-js.md
+  - Tests: 
+    - Tests: ./testing-and-debugging/tests.md
+    - Docker Tests: ./testing-and-debugging/docker-tests.md
+    - Debugging: ./testing-and-debugging/debugging.md
+    - Logger usage: ./testing-and-debugging/logger-usage.md
```
Comment (dutterbutter) [inline docs/mkdocs.yml:50]: Update this to the following:

Test Suite
Debugging

File: docs/mkdocs.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+nav:
+
+  - Getting Started:
+    - Overview: ./index.md
+    - Host Architecture: ./getting-started/overview/host-architecture.md
+    - Package Library: ./getting-started/overview/package-library.md
+    - Installation: ./getting-started/installation.md
+    - General Resources: ./getting-started/resources/general-resources.md
+    - Developer Resources: ./getting-started/resources/developer-resources.md
+  - Usage:
```
Comment (dutterbutter) [inline docs/mkdocs.yml:39]: Can we add a section titled "Running Nodes" and just copy over the three sections outlining how to run nodes from the Installation section?

File: docs/docs/getting-started/overview/package-library.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+- the **trie package** implements a modified merkle-patricia trie.
+
+#### `lib/utils`
+
+- the **utils package** is used to manage node and test directories.
+
+## test packages
+
+#### `?`
```
Comment (dutterbutter) [inline docs/docs/getting-started/overview/package-library.md:119]: Please update accordingly

File: docs/docs/getting-started/overview/host-architecture.md [hunk 8]
```diff
@@ -78,80 +74,16 @@ type host struct {
...
-- `msgSend` - send messages to the network service
-- `msgRec` - receive messages from the network service
-
-within the network service:
-
-- `msgSend` - send messages to the core service
-- `msgRec` - receive messages from the core service
 
-### Syncer Channels
+## Block prodution
```
Comment (dutterbutter) [inline docs/docs/getting-started/overview/host-architecture.md:87]: Mispelled

File: .github/workflows/docs.yml [hunk 9]
```diff
@@ -1,56 +1,23 @@
...
-          git clone --depth=1 --branch="${REMOTE_BRANCH}" --single-branch --no-checkout \
-            "${REMOTE_REPO}" docs/_site/
-      - name: Build site
-        run: bundle exec jekyll build --source docs --destination docs/_site --verbose --trace
+      - name: Checkout main 
+        uses: actions/checkout@v2
+
+      - name: Deploy docs
+        uses: mhausenblas/mkdocs-deploy-gh-pages@master
+        # Or use mhausenblas/mkdocs-deploy-gh-pages@nomaterial to build without the mkdocs-material theme
```
Comment (dutterbutter) [inline .github/workflows/docs.yml:17]: Remove comment

File: docs/mkdocs.yml (hunk 4 above)
Comment (RyRy79261) [reply docs/mkdocs.yml:18]: @dutterbutter 

That feature is for paying users: 
image

Comment (dutterbutter) [reply docs/mkdocs.yml:18]: damn

File: docs/mkdocs.yml (hunk 5 above)
Comment (RyRy79261) [reply docs/mkdocs.yml:50]: @dutterbutter 

Do you mean for me to remove two entries or rename two entries and if so which ones?

Comment (dutterbutter) [reply docs/mkdocs.yml:50]: No need to have 4 individual sections, reduce it to the following

### Testing and Debugging

**Test Suite**
**Debugging**

File: docs/docs/getting-started/overview/package-library.md (hunk 7 above)
Comment (RyRy79261) [reply docs/docs/getting-started/overview/package-library.md:119]: @noot What content do we put here?

File: docs/mkdocs.yml (hunk 5 above)
Comment (RyRy79261) [reply docs/mkdocs.yml]: @dutterbutter Sweet, I dont see content on the debugging page at the moment
https://hackmd.io/bfVYAP2gTe6bjv9TMJ8tqg

File: docs/docs/getting-started/overview/package-library.md (hunk 7 above)
Comment (noot) [reply docs/docs/getting-started/overview/package-library.md:119]: we can add the polkadot.js, rpc, and stress integration tests, just a quick overview is fine

Comment (dutterbutter) [reply docs/docs/getting-started/overview/package-library.md:119]: @RyRy79261 you need to describe the existing test package the can be found [here](https://github.com/ChainSafe/gossamer/tree/development/tests). The same way the above packages are described.

File: docs/mkdocs.yml (hunk 5 above)
Comment (dutterbutter) [reply docs/mkdocs.yml]: As mentioned in standup use the Logger hackmd link

Comment (dutterbutter) [reply docs/mkdocs.yml]: https://hackmd.io/qxsxNhGQQGWYTLfMz5_Oyw

File: docs/docs/getting-started/overview/package-library.md (hunk 7 above)
Comment (RyRy79261) [reply docs/docs/getting-started/overview/package-library.md]: @dutterbutter just updated as described

**Review by noot — COMMENTED** [review]
getting an error with `make build-mkdocs`:
```
$ make build-mkdocs
docker run --rm -it -v /home/elizabeth/go/src/github.com/ChainSafe/gossamer:/docs/docs squidfunk/mkdocs-material build
Unable to find image 'squidfunk/mkdocs-material:latest' locally
latest: Pulling from squidfunk/mkdocs-material
9aae54b2144e: Pull complete 
70269619ad46: Pull complete 
8f02791f0836: Pull complete 
a70ad90f2b5e: Pull complete 
b0c8198636f4: Pull complete 
4f4fb700ef54: Pull complete 
233bd90a8b53: Pull complete 
ae622ab2271e: Pull complete 
f920f6049f8e: Pull complete 
33d90df435f7: Pull complete 
d3daafa05d8b: Pull complete 
063762c42527: Pull complete 
65898433bd03: Pull complete 
6d0321a18334: Pull complete 
Digest: sha256:be1d882064d0a6589b32f37af8e051efe27aabb867f36dd507f56a49a94890d6
Status: Downloaded newer image for squidfunk/mkdocs-material:latest

Config file '/docs/mkdocs.yml' does not exist.
Makefile:132: recipe for target 'build-mkdocs' failed
make: *** [build-mkdocs] Error 1
```

**Comment by arijitAD** [comment]
> @noot Makefile commands dont run on my machine properly so just guessed how it might work there.
> 
> Would maybe need to add PWD to that command to target the yaml file correctly

The command uses a docker image. It should works unless there is some issue with docker.

File: docs/docs/contributing.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+var largeStringTest = []string {
+    "Whoa this test is so big it causes an out of memory issue"
+}
+
+func TestStringTest(t *testing.T) {
+    ...
+}
+
+func TestLargeStringTest(t *testing.T) {
+   	if testing.Short() {
```
Comment (arijitAD) [inline docs/docs/contributing.md:96]: The formatting seems off.

File: docs/docs/contributing.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+```
+$ git remote add gossamer https://github.com/ChainSafe/gossamer.git
+$ git remote -v (you should see myrepo and gossamer in the list of remotes)
+```
+
+**5. Find something to work on.**
+
+To start, check out our open issues. We recommend starting with an [issue labeled `Good First Issue`](https://github.com/ChainSafe/gossamer/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Leave a comment to let us know that you would like to work on it. 
+
+Another option is to improve gossamer where you see fit based on your evaluation of our code. In order to best faciliate collabration, please create an issue before you start working on it.
```
Comment (arijitAD) [inline docs/docs/contributing.md:47]: typo: collaboration
Probably run this doc through grammarly.

File: docs/docs/getting-started/overview/host-architecture.md [hunk 12]
```diff
@@ -78,80 +74,32 @@ type host struct {
...
-- only the **core service** writes to storage state
-- only the **network service** writes to network state
-
-## Node Channels
 
-### Block Channels
+## Block production 
 
-_work in progress_
+<img src="/assets/img/block_production.png" alt="block production" />
```
Comment (arijitAD) [inline docs/docs/getting-started/overview/host-architecture.md:89]: This image doesn't load.

**Comment by noot** [comment]
tried editing the makefile command, but seems like it doesn't use the correct path still:
```
docker run --rm -it -v /home/elizabeth/go/src/github.com/ChainSafe/gossamer/docs/docs squidfunk/mkdocs-material build

Config file '/docs/mkdocs.yml' does not exist.
```

same with using `${PWD}/docs`

is there some other option to specifiy config path?

File: docs/docs/getting-started/overview/host-architecture.md (hunk 12 above)
Comment (RyRy79261) [reply docs/docs/getting-started/overview/host-architecture.md:89]: Unable to replicate
image

File: docs/docs/getting-started/resources/developer-resources.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
+---
+layout: default
+title: Developer Resources
+permalink: /developer-resources/
+---
+
+### Specification
+
+- <a target="_blank" rel="noopener noreferrer" href="https://github.com/w3f/polkadot-spec/blob/master/polkadot-host-spec/polkadot_host_spec.pdf">Polkadot Host Specification</a>
```
[the remark this answers is not in the document]
Comment (dutterbutter) [reply docs/docs/getting-started/resources/developer-resources.md:9]: You can just use: https://github.com/w3f/polkadot-spec/tree/master/host-spec

**Review by dutterbutter — APPROVED** [review]

**Review by edwardmack — APPROVED** [review]

**Pull request merged**